caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditConnection.java

```diff
@@ -21,18 +21,46 @@
 import org.elasticsearch.client.RestHighLevelClient;
 
 import java.io.IOException;
+import java.util.Arrays;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class ElasticAuditConnection implements AuditConnection {
+     private static final Logger LOG = LogManager.getLogger(ElasticAuditConnection.class.getName());
 
     private final RestHighLevelClient restHighLevelClient;
     private ElasticAuditIndexManager indexManager;
 
     public ElasticAuditConnection() throws AuditConfigurationException
     {
-            final String hostAndPorts = 
+            String hostAndPorts = 
                 System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES,
                                    System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES));
-
+            final String hostValues = 
+                System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_VALUES,
+                                   System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_VALUES));
+            final String port = 
+                System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_PORT_VALUE,
+                                   System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_PORT_VALUE));
+            final StringBuilder hostAndPortBuilder = new StringBuilder("");
+            final String[] hostArray;
+            
+            if(hostAndPorts == null){
+                hostArray = hostValues.split(",");
+                if (hostArray.length == 0) {
+                    final String errorMessage = "No hosts configured.";
+                    LOG.error(errorMessage);
+                    throw new AuditConfigurationException(errorMessage);
+                }
+                for (int index = 0; index < hostArray.length; index++){
+                    hostAndPortBuilder.append(hostArray[index] + ":" + port + ",");
```

Comment:
since you are already using a string builder, use append instead of string concatenation and single quotes for single character